crypto/src/hash/pedersen/mod.rs

```diff
@@ -0,0 +1,102 @@
+use lambdaworks_math::{
+    cyclic_group::IsGroup,
+    elliptic_curve::short_weierstrass::{
+        curves::stark_curve::StarkCurve, point::ShortWeierstrassProjectivePoint,
+        traits::IsShortWeierstrass,
+    },
+    field::{
+        element::FieldElement, fields::fft_friendly::stark_252_prime_field::Stark252PrimeField,
+    },
+};
+
+pub mod parameters;
+use self::parameters::PedersenParameters;
+
+pub struct Pedersen<EC: IsShortWeierstrass> {
+    params: PedersenParameters<EC>,
+}
```

Comment:
The generic when it only works for one curve seems a bit premature. But it's ok.